pulumi/infra/ux_router.py

```diff
@@ -0,0 +1,50 @@
+from typing import Optional
+
+import pulumi_aws as aws
+from infra.bucket import Bucket
+from infra.config import GLOBAL_LAMBDA_ZIP_TAG
+from infra.lambda_ import Lambda, LambdaExecutionRole, PythonLambdaArgs, code_path_for
+from infra.metric_forwarder import MetricForwarder
+from infra.network import Network
+from infra.secret import JWTSecret
+
+import pulumi
+
+
+class UxRouter(pulumi.ComponentResource):
+    def __init__(
+        self,
+        network: Network,
+        secret: JWTSecret,
+        ux_bucket: Bucket,
+        forwarder: Optional[MetricForwarder] = None,
+        opts: Optional[pulumi.ResourceOptions] = None,
+    ) -> None:
+
+        name = "ux-router"
+        super().__init__("grapl:UXRouter", name, None, opts)
+
+        self.role = LambdaExecutionRole(name, opts=pulumi.ResourceOptions(parent=self))
+
+        self.function = Lambda(
+            name,
+            args=PythonLambdaArgs(
+                execution_role=self.role,
+                description=GLOBAL_LAMBDA_ZIP_TAG,
+                handler="lambdex_handler.handler",
+                code_path=code_path_for(name),
+                env={"GRAPL_LOG_LEVEL": "INFO", "UX_BUCKET_NAME": ux_bucket.bucket},
+                timeout=5,
+                memory_size=128,
+            ),
+            # TODO: I don't think we need a network, because I don't
```

Comment:
ux-router only acts as a (bad) proxy to S3. Not sure if this mean sit does or doesn't need `network`, but yeah.

Reply:
Yeah, I'm keeping consistent with what's in the CDK code as much as possible. There are a handful of places like this and a few others where it looks like we're doing more than is strictly necessary. Once we're transitioned, I'm going to make another pass for those kinds of things.